### PR TITLE
Added a test to check changes made to the table after creating it and committing it

### PR DESCRIPTION
**Pull request by fedch**
## Description

Added triggers to add-dataset command for tracking changes

Add-dataset command lacked proper tracking triggers for tables, causing changes to be undetected.

Fixed by ensuring the _create_triggers function is called during the commit process within the same database transaction as reading the diff, preventing missed rows during commit.

## Related links:

[805](https://github.com/koordinates/kart/issues/805)

## Checklist:

- [X] Have you reviewed your own change?
- [X] Have you included test(s)?
- [X] Have you updated the [changelog](https://github.com/koordinates/kart/blob/master/CHANGELOG.md)?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,8 @@ _When adding new entries to the changelog, please include issue/PR numbers where
 
 - Added support for `--diff-format` option under `kart show` command to display different level of detail depending on the user's needs.  [#721](https://github.com/koordinates/kart/issues/721)
 
+- Resolved issue [#805](https://github.com/koordinates/kart/issues/805) by ensuring the `add-dataset` command creates tracking triggers for committed tables, enabling future changes to be detected and committed.
```
Comment (craigds) [inline CHANGELOG.md:16]: add-dataset is a new feature in this release (L10) so we don't need to mention this - the whole feature is new

File: kart/tabular/working_copy/mysql.py [hunk 2]
```diff
@@ -160,6 +160,10 @@ def _sno_tracking_name(self, trigger_type, dataset=None):
         """Returns the sno-branded name of the trigger reponsible for populating the sno_track table."""
         return f"_sno_track_{trigger_type}"
 
+    def create_triggers(self, sess, dataset):
```
Comment (craigds) [inline kart/tabular/working_copy/mysql.py:163]: I'd suggest just renaming the original method, no point having two

File: kart/add_dataset.py [hunk 3]
```diff
@@ -69,6 +73,14 @@ def add_dataset(ctx, table_name, message, launch_editor, output_format):
         table_name, meta_items
     )
 
+    # Create a table object from the table_name:
+    path = TableV3.dataset_path_to_table_name(table_name)
```
Comment (craigds) [inline kart/add_dataset.py:77]: I'm not sure what this line adds - table_name is already a table name so it seems like this line converts it the wrong way and basically isn't required?

File: kart/add_dataset.py [hunk 4]
```diff
@@ -69,6 +73,14 @@ def add_dataset(ctx, table_name, message, launch_editor, output_format):
...
+    table = TableV3.new_dataset_for_writing(path, meta_items["schema.json"], repo)
+
+    # Add triggers on a new table if they aren't already present:
+    with wc.session() as sess:
```
Comment (craigds) [inline kart/add_dataset.py:81]: there's a race condition we should think about:

* user runs `kart add-dataset` on a table with 10 rows
* we build a diff
* user inserts a new row
* then we create triggers
* then we commit 10 rows to the dataset

This sequence of events causes the triggers to miss an insert - the row is not tracked by the tracking table, nor is it committed. User won't see a diff until they change the row and it gets picked up by the trigger again (it will be considered an update rather than an insert, even though the row hasn't been committed)

What if instead, we created the triggers before we fetch the diff?

* user runs `kart add-dataset` on a table with 10 rows
* we create triggers
* user inserts a new row
* trigger tracks a new inserted row
* we fetch a diff of the whole table
* we commit the diff

In this situation^ we have a different issue - the row is _both_ committed and in the tracking table - in both cases as an 'insert'. I'm not sure if this causes problems or not. I'd hope it doesn't show up in `kart diff` but I might be mistaken.


@olsen232 is this something we should address?

Comment (olsen232) [reply kart/add_dataset.py:81]: Having unneeded primary-key-values in the tracking table isn't that bad - they don't cause the diffs to contain any extra deltas, they just slightly add to Kart's workload (in the extreme case, we could put every row in the tracking table for no reason, kart would still work fine, but it would lose the efficiency that the tracking table provides). So, creating triggers early is preferable to creating them late.

Apart from that preference, I wouldn't put too much more thought into it... all of this probably happens inside a single transaction, so, I don't know if reordering the pieces of it will make any difference, or if it all appears atomic to another editor. We could test it further but it seems like a lot of work for a weird edge case

**Review by craigds — APPROVED** [review]
